### PR TITLE
create quantum_fourier_transform

### PR DESCRIPTION
**Pull request by KevinJoven11**
This is part of the #Hacktoberfest.
I build the quantum fourier transform for N qubits. (n = 3 in the example) Best,
Kevin

### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by KevinJoven11** [comment]
@cclauss Hi! Sorry for ping you. I just wanna know if this repo (Algorithms/Python) is not more part of the #Hacktoberfest or is there another reason for that. I just wanna have my second contribution to the hackfest to have the t-shirt.
Best,

**Comment by cclauss** [comment]
We had a handful of pull requests when October started and we have almost 200 now so I view Hacktoberfest to be a bit of a denial of service attack on my productivity but we do give out the labels (added here) for pull requests that meet out CONTRIBUTING.md requirements.

A function name like `qft()` is not going to be acceptable.  Also we require doctests.

File: quantum/q_fourier_transform.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+https://en.wikipedia.org/wiki/Quantum_Fourier_transform
+https://qiskit.org/textbook/ch-algorithms/quantum-fourier-transform.html
+"""
+
+import numpy as np
+import qiskit
+from qiskit import Aer, ClassicalRegister, QuantumCircuit, QuantumRegister, execute
+
+
+def qft(n: int = 3) -> qiskit.result.counts.Counts:
```
Comment (cclauss) [inline quantum/q_fourier_transform.py:19]:
```suggestion
def quantum_fourier_transform(number: int = 3) -> qiskit.result.counts.Counts:
```

File: quantum/q_fourier_transform.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    qr_1: ──────┼────────■───────┤ H ├─■─────────────┼─
+          ┌───┐ │P(π/4)  │P(π/2) └───┘               │
+    qr_2: ┤ H ├─■────────■───────────────────────────X─
+          └───┘
+    cr: 3/═════════════════════════════════════════════
+    Args:
+        n : number of qubits
+    Returns:
+        qiskit.result.counts.Counts: Counts for each state.
+    """
```
Comment (cclauss) [inline quantum/q_fourier_transform.py:37]: Let’s add a doctest that will fail…. Please add a few more with zero, negative numbers, floating point numbers, letters, etc.  We want to see how the function runs and how if fails on garbage inputs.
```suggestion
    >>> quantum_fourier_transform(77)
    66
    """
```

File: quantum/q_fourier_transform.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    cr = ClassicalRegister(n, "cr")
+
+    quantum_circuit = QuantumCircuit(qr, cr)
+
+    counter = n
+    # Add the hadamard gates and the CROT gates.
+    for i in range(counter):
+
+        quantum_circuit.h(n - i - 1)
+        counter = counter - 1
```
Comment (cclauss) [inline quantum/q_fourier_transform.py:48]:
```suggestion
        counter -= 1
```

File: quantum/q_fourier_transform.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    quantum_circuit.measure(qr, cr)
+    # simulate with 10000 shots
+    backend = Aer.get_backend("qasm_simulator")
+    job = execute(quantum_circuit, backend, shots=10000)
+
+    return job.result().get_counts(quantum_circuit)
+
+
+if __name__ == "__main__":
+    print(f"Total count for qft state is: {qft(3)}")
```
Comment (cclauss) [inline quantum/q_fourier_transform.py:65]:
```suggestion
    print(f"Total count for quantum fourier transform state is: {qft(3)}")
```

**Comment by cclauss** [comment]
Extra credit:  In a ___separate pull request___, fix the `PendingDeprecationWarning` that pytest is raising in our automated testing… https://github.com/TheAlgorithms/Python/actions/runs/3254565205/jobs/5342967879#step:6:644

File: quantum/q_fourier_transform.py [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+          ┌───┐ │P(π/4)  │P(π/2) └───┘               │
+    qr_2: ┤ H ├─■────────■───────────────────────────X─
+          └───┘
+    cr: 3/═════════════════════════════════════════════
+    Args:
+        n : number of qubits
+    Returns:
+        qiskit.result.counts.Counts: distribute counts.
+
+    >>> quantum_fourier_transform(-1)
```
Comment (cclauss) [inline quantum/q_fourier_transform.py:39]: Can we please get at least on doctest that passes?

**Review by cclauss — APPROVED** [review]

**Pull request merged**